.travis/common.sh

```diff
@@ -66,25 +66,28 @@ then
   exit 0;
 fi
 
-# variables QUAY_USERNAME and QUAY_PASSWORD are required to push docker images
-if [ "$QUAY_USERNAME" == "" ];
+# variables DOCKER_USERNAME and DOCKER_PASSWORD are required to push docker images
+if [ "$DOCKER_USERNAME" == "" ];
 then
-  echo "Environment variable QUAY_USERNAME is missing.";
+  echo "Environment variable DOCKER_USERNAME is missing.";
   exit 0;
 fi
 
-if [ "$QUAY_PASSWORD" == "" ];
+if [ "$DOCKER_PASSWORD" == "" ];
 then
-  echo "Environment variable QUAY_PASSWORD is missing.";
+  echo "Environment variable DOCKER_PASSWORD is missing.";
   exit 0;
 fi
 
 function docker_tag_exists() {
     TAG=${2//\"/}
-    IMAGES=$(curl -s -H "Authorization: Bearer ${QUAY_PASSWORD}" https://quay.io/api/v1/repository/$1-$3/image/ | jq '.images | sort_by(.sort_index) | .[] .tags | select(.[] !=null) | .[0]' | sed s/\"//g)
-    if echo "$IMAGES" | grep -q -P "(^|\s)$TAG(?=\s|$)" ; then
-        return 0
+    TOKEN=$( curl -sSLd "username=${DOCKER_USERNAME}&password=${DOCKER_PASSWORD}" https://hub.docker.com/v2/users/login | jq -r ".token" )
+    RES=$(curl -sH "Authorization: JWT $TOKEN" "https://hub.docker.com/v2/repositories/shopify/$1-$3/tags/$TAG/" | jq .detail)
+
+    if [ "$RES" == "\"Not found\"" ];
+    then
+        return 1
```

Comment:
hmm this does not read right - so it says if image with given tag not found then return 1 which means tag exists?

Reply:
Hmm. I followed the previous logic of: 
```
if echo "$IMAGES" | grep -q -P "(^|\s)$TAG(?=\s|$)" ; then
        return 0
    fi

    return 1
```

Reply:
I interpreted the if condition checking if the tag exists, and if so return 0. Otherwise return 1?

Reply:
so TIL
```
> ingress-nginx (master)$ cat sh
#!/usr/bin/env bash

function docker_tag_exists() {
  return 0
}

if docker_tag_exists;
then
  echo "yes 0 is truthy"
fi
> ingress-nginx (master)$ bash sh
yes 0 is truthy
```